File: python-wrapper/src/python/model/embed/DocumentType.py
from typing import Union, Dict
from enum import Enum

class DocumentType(Enum):
	Admission = "admission"
	Alert = "alert"
	BvtSample = "bvt_sample"
	Clinicalpath = "clinicalpath"
	Clinicalsummary = "clinicalsummary"
	Contactreport = "contactreport"
	Quote = "quote"
	Invoice = "invoice"
	Death = "death"
	Discharge = "discharge"
	Dischargereport = "dischargereport"
	EbirthBabyMedicalform = "ebirth_baby_medicalform"
	EbirthBabyNotification = "ebirth_baby_notification"
	EbirthMotherMedicalform = "ebirth_mother_medicalform"
	EbirthMotherNotification = "ebirth_mother_notification"
	EcareSafeConsultation = "ecare_safe_consultation"
	Epidemiology = "epidemiology"
	Intervention = "intervention"
	Labrequest = "labrequest"
	Labresult = "labresult"
	Medicaladvisoragreement = "medicaladvisoragreement"
	Medicationschemeelement = "medicationschemeelement"
	Note = "note"
	Notification = "notification"
	Pharmaceuticalprescription = "pharmaceuticalprescription"
	Prescription = "prescription"
	Productdelivery = "productdelivery"
	Quickdischargereport = "quickdischargereport"
	Radiationexposuremonitoring = "radiationexposuremonitoring"
	Referral = "referral"
	Report = "report"
	Request = "request"
	Result = "result"
	Sumehr = "sumehr"
	Telemonitoring = "telemonitoring"
	Template = "template"
	TemplateAdmin = "template_admin"
	Treatmentsuspension = "treatmentsuspension"
	Vaccination = "vaccination"

	def __serialize__(self) -> object:
		return self.value

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'DocumentType':
		if data == "admission":
			return DocumentType.Admission
		elif data == "alert":
			return DocumentType.Alert
		elif data == "bvt_sample":
			return DocumentType.BvtSample
		elif data == "clinicalpath":
			return DocumentType.Clinicalpath
		elif data == "clinicalsummary":
			return DocumentType.Clinicalsummary
		elif data == "contactreport":
			return DocumentType.Contactreport
		elif data == "quote":
			return DocumentType.Quote
		elif data == "invoice":
			return DocumentType.Invoice
		elif data == "death":
			return DocumentType.Death
		elif data == "discharge":
			return DocumentType.Discharge
		elif data == "dischargereport":
			return DocumentType.Dischargereport
		elif data == "ebirth_baby_medicalform":
			return DocumentType.EbirthBabyMedicalform
		elif data == "ebirth_baby_notification":
			return DocumentType.EbirthBabyNotification
		elif data == "ebirth_mother_medicalform":
			return DocumentType.EbirthMotherMedicalform
		elif data == "ebirth_mother_notification":
			return DocumentType.EbirthMotherNotification
		elif data == "ecare_safe_consultation":
			return DocumentType.EcareSafeConsultation
		elif data == "epidemiology":
			return DocumentType.Epidemiology
		elif data == "intervention":
			return DocumentType.Intervention
		elif data == "labrequest":
			return DocumentType.Labrequest
		elif data == "labresult":
			return DocumentType.Labresult
		elif data == "medicaladvisoragreement":
			return DocumentType.Medicaladvisoragreement
		elif data == "medicationschemeelement":
			return DocumentType.Medicationschemeelement
		elif data == "note":
			return DocumentType.Note
		elif data == "notification":
			return DocumentType.Notification
		elif data == "pharmaceuticalprescription":
			return DocumentType.Pharmaceuticalprescription
		elif data == "prescription":
			return DocumentType.Prescription
		elif data == "productdelivery":
			return DocumentType.Productdelivery
		elif data == "quickdischargereport":
			return DocumentType.Quickdischargereport
		elif data == "radiationexposuremonitoring":
			return DocumentType.Radiationexposuremonitoring
		elif data == "referral":
			return DocumentType.Referral
		elif data == "report":
			return DocumentType.Report
		elif data == "request":
			return DocumentType.Request
		elif data == "result":
			return DocumentType.Result
		elif data == "sumehr":
			return DocumentType.Sumehr
		elif data == "telemonitoring":
			return DocumentType.Telemonitoring
		elif data == "template":
			return DocumentType.Template
		elif data == "template_admin":
			return DocumentType.TemplateAdmin
		elif data == "treatmentsuspension":
			return DocumentType.Treatmentsuspension
		elif data == "vaccination":
			return DocumentType.Vaccination
		else:
			raise Exception(f"{data} is not a valid value for DocumentType enum.")

__all__ = ['DocumentType']

File: cardinal-sdk/src/commonMain/kotlin/com/icure/cardinal/sdk/model/embed/DocumentType.kt
package com.icure.cardinal.sdk.model.embed

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.String

@Serializable
public enum class DocumentType(
	internal val dtoSerialName: String,
) {
	@SerialName("admission")
	Admission("admission"),

	@SerialName("alert")
	Alert("alert"),

	@SerialName("bvt_sample")
	BvtSample("bvt_sample"),

	@SerialName("clinicalpath")
	Clinicalpath("clinicalpath"),

	@SerialName("clinicalsummary")
	Clinicalsummary("clinicalsummary"),

	@SerialName("contactreport")
	Contactreport("contactreport"),

	@SerialName("quote")
	Quote("quote"),

	@SerialName("invoice")
	Invoice("invoice"),

	@SerialName("death")
	Death("death"),

	@SerialName("discharge")
	Discharge("discharge"),

	@SerialName("dischargereport")
	Dischargereport("dischargereport"),

	@SerialName("ebirth_baby_medicalform")
	EbirthBabyMedicalform("ebirth_baby_medicalform"),

	@SerialName("ebirth_baby_notification")
	EbirthBabyNotification("ebirth_baby_notification"),

	@SerialName("ebirth_mother_medicalform")
	EbirthMotherMedicalform("ebirth_mother_medicalform"),

	@SerialName("ebirth_mother_notification")
	EbirthMotherNotification("ebirth_mother_notification"),

	@SerialName("ecare_safe_consultation")
	EcareSafeConsultation("ecare_safe_consultation"),

	@SerialName("epidemiology")
	Epidemiology("epidemiology"),

	@SerialName("intervention")
	Intervention("intervention"),

	@SerialName("labrequest")
	Labrequest("labrequest"),

	@SerialName("labresult")
	Labresult("labresult"),

	@SerialName("medicaladvisoragreement")
	Medicaladvisoragreement("medicaladvisoragreement"),

	@SerialName("medicationschemeelement")
	Medicationschemeelement("medicationschemeelement"),

	@SerialName("note")
	Note("note"),

	@SerialName("notification")
	Notification("notification"),

	@SerialName("pharmaceuticalprescription")
	Pharmaceuticalprescription("pharmaceuticalprescription"),

	@SerialName("prescription")
	Prescription("prescription"),

	@SerialName("productdelivery")
	Productdelivery("productdelivery"),

	@SerialName("quickdischargereport")
	Quickdischargereport("quickdischargereport"),

	@SerialName("radiationexposuremonitoring")
	Radiationexposuremonitoring("radiationexposuremonitoring"),

	@SerialName("referral")
	Referral("referral"),

	@SerialName("report")
	Report("report"),

	@SerialName("request")
	Request("request"),

	@SerialName("result")
	Result("result"),

	@SerialName("sumehr")
	Sumehr("sumehr"),

	@SerialName("telemonitoring")
	Telemonitoring("telemonitoring"),

	@SerialName("template")
	Template("template"),

	@SerialName("template_admin")
	TemplateAdmin("template_admin"),

	@SerialName("treatmentsuspension")
	Treatmentsuspension("treatmentsuspension"),

	@SerialName("vaccination")
	Vaccination("vaccination"),
}
